src/goto-symex/builtin_functions.cpp

```diff
@@ -327,8 +327,18 @@ void goto_symext::symex_printf(const expr2tc &lhs, expr2tc &rhs)
 
   code_printf2t &new_rhs = to_code_printf2t(renamed_rhs);
 
-  // The expr2tc in position 0 is the string format
-  const irep_idt fmt = get_string_argument(new_rhs.operands[0]);
+  // get string format
+  while(!is_constant_string2t(get_base_object(new_rhs.operands[0])))
```

Comment:
This is unlikely to work. You will have to make sure that you *know* where the format string is. I.e. it's already known:
- printf: 1st argument
- fprintf, sprintf, dprintf: 2nd argument
- snprintf: 3rd argument

Unfortunately we can't even be sure that the format string going to be a constant-string expression, it could well be a symbol. I suppose the easiest approach to cover all these cases is to give `symex_printf` another `int` parameter saying which argument will be the format string. If it is not a constant-string expression, there's not a lot we can do, I guess. But the important point is: searching until we find a constant-string won't guarantee that this actually is the format string.

Reply:
changed as suggested